bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/scheduler2/CronAdjuster.java

```diff
@@ -0,0 +1,551 @@
+        int[] r = new int[2];
+        r[0] = 0;
+        r[1] = max;
+        if ("*".equals(range)) {
+            return r;
+        }
+
+        String parts[] = range.split("-");
+        r[0] = r[1] = parseInt(parts[0], min, max, names);
+        if (parts.length == 2) {
+            r[1] = parseInt(parts[1], min, max, names);
+        }
+
+        if (r[0] < min) {
+            throw new IllegalArgumentException("Value too small: " + r[0] + " for " + cf.toString());
+        }
+        if (r[1] > max) {
+            throw new IllegalArgumentException("Value too high: " + r[1] + " for " + cf.toString());
+        }
+
+        return r;
+    }
+
+    private int parseInt(String string, int min, int max, String[] names) {
+        if (string.isEmpty()) {
+            return 0;
+        }
+
+        for (int n = 0; n < names.length; n++) {
+            if (names[n].equals(string)) {
+                return n + min;
+            }
+        }
+
+        return Integer.parseInt(string);
+    }
+
+    @Override
+    public Temporal adjustInto(Temporal temporal) {
+
+        //
+        // Never match the actual time, so since our basic
+        // unit is seconds, we add one second.
+        //
+
+        Temporal ret = temporal.plus(1, ChronoUnit.SECONDS);
+
+        //
+        // We loop through the fields until they all match. If
+        // one of them does not match, its type is incremented
+        // and all lower fields are reset to their minimum. And
+        // we start over with this new time.
+        //
+
+        loop: while (true) {
+
+            for (Field field : fields) {
+                if (field != null) {
+                    Temporal out = field.isOk(ret);
+                    if (out != null) {
+                        ret = out;
+                        continue loop;
+                    }
+                }
+            }
+
+            //
+            // All fields match!
+            //
+
+            return ret;
+        }
+    }
+
+    /*
+     * Helper to create an or expression of 2 checkers.
+     */
+    private Checker or(Checker a, Checker b) {
+        if (a == ALWAYS_TRUE || b == ALWAYS_TRUE) {
+            return ALWAYS_TRUE;
+        }
+
+        if (a == ALWAYS_TRUE) {
+            return b;
+        }
+
+        if (b == ALWAYS_TRUE) {
+            return a;
+        }
+
+        return (temporal) -> a.matches(temporal) || b.matches(temporal);
+    }
+
+    /*
+     * Helper to create an and expression of 2 checkers.
+     */
+    private Checker and(Checker a, Checker b) {
+        if (a == ALWAYS_FALSE || b == ALWAYS_FALSE) {
+            return ALWAYS_FALSE;
+        }
+
+        if (a == ALWAYS_TRUE) {
+            return b;
+        }
+
+        if (b == ALWAYS_TRUE) {
+            return a;
+        }
+
+        return (temporal) -> a.matches(temporal) && b.matches(temporal);
+    }
+
+    public Map<String, String> getEnv() {
```

Comment:
How the map returned by `getEnv()` be used? It doesn't seem to be used anywhere and CronAdjuster is a package private class. Is this something to pass to the `run` method in `CronJob`? or should it be made available somewhere else?

Reply:
If you can't find any usages, then there probably aren't any. I just haven't come to cleaning up the adjusters, so potentially they are not needed. I would try to keep the API as slim as possible, therefore feel free to dump everything which appears to be not needed.